File: instrumentation/vertx-core-4.5.1/src/main/java/com/nr/vertx/instrumentation/OutboundWrapper.java
/*
 *
 *  * Copyright 2024 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.vertx.instrumentation;

import com.newrelic.api.agent.HeaderType;
import com.newrelic.api.agent.OutboundHeaders;
import io.vertx.core.MultiMap;

public class OutboundWrapper implements OutboundHeaders {

    private final MultiMap headers;

    public OutboundWrapper(MultiMap headers) {
        this.headers = headers;
    }

    @Override
    public HeaderType getHeaderType() {
        return HeaderType.HTTP;
    }

    @Override
    public void setHeader(String name, String value) {
        headers.add(name, value);
    }

}

File: instrumentation/akka-http-2.13_10.2.0/src/main/scala/com/agent/instrumentation/akka/http/OutboundWrapper.scala
/*
 *
 *  * Copyright 2020 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.agent.instrumentation.akka.http

import akka.http.scaladsl.model.HttpResponse
import akka.http.scaladsl.model.headers.RawHeader
import com.newrelic.api.agent.{HeaderType, OutboundHeaders}

class OutboundWrapper(var response: HttpResponse) extends OutboundHeaders {

  def getHeaderType: HeaderType = {
    HeaderType.HTTP
  }

  def setHeader(name: String, value: String): Unit = {
    response = response.addHeader(new RawHeader(name, value))
  }

  def getHeader(name: String): String = {
    response.headers.find(header => header.is(name.toLowerCase)).map(header => header.value).orNull
  }

}
